daemon/metrics.go

```diff
@@ -0,0 +1,84 @@
+package daemon
+
+import (
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/dockerversion"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/go-metrics"
+)
+
+var (
+	containerActions   metrics.LabeledTimer
+	imageActions       metrics.LabeledTimer
+	containerCounter   metrics.LabeledCounter
+	daemonActions      metrics.LabeledTimer
+	networkActions     metrics.LabeledTimer
+	engineVersion      metrics.LabeledGauge
+	healthCheckCounter metrics.Counter
+)
+
+func init() {
+	ns := metrics.NewNamespace("engine", "daemon", metrics.Labels{
+		"version": dockerversion.Version,
+		"commit":  dockerversion.GitCommit,
+	})
+	containerActions = ns.NewLabeledTimer("container_actions", "The number of seconds it takes to process each container action", "action")
+	daemonActions = ns.NewLabeledTimer("daemon_actions", "The number of seconds it takes to process each daemon action", "action")
+	networkActions = ns.NewLabeledTimer("network_actions", "The number of seconds it takes to process each network action", "action")
+	engineVersion = ns.NewLabeledGauge("version", "The version and commit information for the engine process", metrics.Unit("info"),
+		"architecture",
+		"cpus",
+		"graph_driver",
+		"kernel",
+		"memory",
+		"os",
+	)
+	containerCounter = ns.NewLabeledCounter("container", "The total number of containers processed", "action")
+	healthCheckCounter = ns.NewCounter("health_check", "The total number of health checks executed")
+	imageActions = ns.NewLabeledTimer("image_actions", "The number of seconds it takes to process each image action", "action")
+	metrics.Register(ns)
+}
+
+func (daemon *Daemon) startMetrics(c *container.Container) {
+	ns := metrics.NewNamespace("container", "stat", metrics.Labels{
+		"id":    c.ID,
+		"image": c.Config.Image,
+	})
+
+	var (
+		cpuTime          = ns.NewLabeledGauge("cpu_usage", "The number of nanoseconds a container uses in cpu time", metrics.Nanoseconds, "type")
+		memoryUsage      = ns.NewLabeledGauge("memory_usage", "The number of bytes a container uses in memory", metrics.Bytes, "type")
+		memoryFailCount  = ns.NewGauge("memory_failcount", "The number of times the container has reached the memory limit", metrics.Total)
+		pidCount         = ns.NewLabeledGauge("pids", "The number of pids running inside a container", metrics.Total, "type")
+		networkRxPackets = ns.NewLabeledGauge("network_rx", "The number of packets received by the container", metrics.Total, "type", "interface")
+		networkTxPackets = ns.NewLabeledGauge("network_tx", "The number of packets transmitted by the container", metrics.Total, "type", "interface")
+		networkRxBytes   = ns.NewLabeledGauge("network_rx", "The number of bytes received by the container", metrics.Bytes, "interface")
+		networkTxBytes   = ns.NewLabeledGauge("network_tx", "The number of bytes transmitted by the container", metrics.Bytes, "interface")
+	)
+	metrics.Register(ns)
+
+	update := daemon.subscribeToContainerStats(c)
```

Comment:
Sounds like a good idea.  I was trying to keep the scope small at this point to make sure we have the basics monitored before adding more metrics.  This is just the first metrics PR, we will expand the scope after to other packages/etc.